File: 2017/01/JavaScript/captcha.js
#!/usr/bin/env js

"use strict";

let r = require('readline').createInterface({input: process.stdin, output: process.stdout})

r.question("", function(inp) {
    console.log("P1: ",inp.split('').reduce((s, v, i, a) => parseInt(s) + ( (v == a[(i+1) % a.length]) ? parseInt(v) : 0 ),0));
    console.log("P2: ",inp.split('').reduce((s, v, i, a) => parseInt(s) + ( (v == a[(i+a.length/2) % a.length]) ? parseInt(v) : 0 ),0));
    r.close();
});

/*****************************
* Author: Rahul Butani       *
* Date:   December 1st, 2017 *
*****************************/

File: 2017/01/Scala/captcha.scala
val input = scala.io.StdIn.readLine().split("").map(_.toInt)

println(s"P1: ${(input :+ input.head).sliding(2).toList.foldLeft(0)((s, t) => (if (t(0) == t(1)) t(0) + s else s))}") 
println(s"P2: ${(input zip (input.drop(input.length/2) ++ input.take(input.length/2))).foldLeft(0)((s, t) => if (t._1 == t._2) t._1 + s else s)}")

/*****************************
* Author: Rahul Butani       *
* Date:   December 2nd, 2017 *
*****************************/
